src/fs/metadata_log.cc

```diff
@@ -89,6 +94,17 @@ future<> metadata_log::shutdown() {
     });
 }
 
+void metadata_log::throw_if_read_only_fs() {
+    if (__builtin_expect(_read_only_fs == read_only_fs::yes, false)) {
+        throw read_only_filesystem_exception();
+    }
+}
+
+void metadata_log::set_fs_read_only_mode(read_only_fs val) noexcept {
```

Comment:
I'm not sure if logger::warn() does not throw.

Reply:
logger::warn() calls logger::log() which looks like this
```c++
template <typename... Args>
    void log(log_level level, const char* fmt, const Args&... args) {
        if (is_enabled(level)) {
            try {
                do_log(level, fmt, args...);
            } catch (...) {
                failed_to_log(std::current_exception());
            }
        }
    }
```
is_enabled():
```c++
bool is_enabled(log_level level) const {
    return __builtin_expect(level <= _level.load(std::memory_order_relaxed), false);
}
```
_level is std::atomic and std::atomic::load() is noexcept.
logger::failed_to_log() looks like this
```c++
void logger::failed_to_log(std::exception_ptr ex)
{
    try {
        do_log(log_level::error, "failed to log message: {}", ex);
    } catch (...) {
        ++logging_failures;
    }
}
```
copy constructor of exception_ptr is noexcept.
So I think that current implementation of logger::warn is noexcept after all.

Reply:
Ok, it can be `noexcept`. I wonder why seastar does not declare all these functions as `noexcept` though.